server/http/controller.go

```diff
@@ -483,6 +485,20 @@ func metadata(in *map[string]string) model.RunMetadata {
 	return model.RunMetadata(*in)
 }
 
+func environment(ctx context.Context, testDB model.Repository, environmentId string) model.Environment {
+	if environmentId != "" {
+		environment, err := testDB.GetEnvironment(ctx, environmentId)
+
+		if err != nil {
+			return model.Environment{}
+		}
```

Comment:
We have a lot of different errors that can happen here (invalid SQL syntax, connection errors, etc) that we need to handle. I'd suggest adding an error return as well, but that would totally kill the reason for this function to exist. Maybe that logic could be done in the `RunTest` function instead. So you can handle it using `handleDBError`

Reply:
I added the error handling, let me know if it makes sense